inc/modules/export/epub/class-epub.php

```diff
@@ -527,20 +546,18 @@ public function convertGenerator(): \Generator {
 	 * @return mixed
 	 */
 	protected function preProcessBookContents( $book_contents ) {
-
 		// We need to change global $id for shortcodes, the_content, ...
 		global $id;
 		$old_id = $id;
 
 		// Do root level structures first.
 		foreach ( $book_contents as $type => $struct ) {
 
-			if ( preg_match( '/^__/', $type ) ) {
+			if ( 0 === strpos( $type, '__' ) ) {
```

Comment:
Regular expressions are kind of hard to read, once we move to PHP 8 this could be replaced by `str_starts_with`

Reply:
I also swapped those because the regex that we are performing is less performant for a simple string lookup